internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,199 @@
+			},
+		},
+	}
+	return &v1.DaemonSet{
+
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        dsName,
+			Namespace:   namespace,
+			Annotations: dsAnnotations,
+		},
+		Spec: v1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: matchLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: matchLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers:       []corev1.Container{container},
+					PreemptionPolicy: &preempt,
+					Priority:         &zeroInt32,
+					HostNetwork:      true,
+					Tolerations: []corev1.Toleration{
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/not-ready",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/unreachable",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect: "NoSchedule",
+							Key:    "node-role.kubernetes.io/master",
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "host",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+									Type: &hostPathType,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// Delete daemon set
+func DeleteDaemonSet(daemonSetName, namespace string) error {
+	tnf.ClaimFilePrintf("Deleting daemon set %s\n", daemonSetName)
+	deletePolicy := metav1.DeletePropagationForeground
+	client := clientsholder.GetClientsHolder().K8sClient.AppsV1()
+	if err := client.DaemonSets(namespace).Delete(context.TODO(), daemonSetName, metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
+	}); err != nil {
+		tnf.ClaimFilePrintf("The daemonset (%s) deletion is unsuccessful due to %s", daemonSetName, err)
+	}
+	doneCleanUp := false
+	for start := time.Now(); !doneCleanUp && time.Since(start) < timeout; {
+		client := clientsholder.GetClientsHolder().K8sClient
+		pods, err := client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "name=" + daemonSetName})
+		if err != nil {
+			return fmt.Errorf("failed to get pods, err: %s", err)
+		}
+		//nolint:ineffassign
+		if len(pods.Items) == 0 {
+			doneCleanUp = true
+			break
+		}
+		time.Sleep(time.Duration(timeout.Minutes()))
+	}
+
+	tnf.ClaimFilePrintf("Successfully cleaned up daemon set %s\n", daemonSetName)
+	return nil
+}
+
+// Check if the daemon set exists
+func doesDaemonSetExist(daemonSetName, namespace string) bool {
+	client := clientsholder.GetClientsHolder().K8sClient.AppsV1()
+	_, err := client.DaemonSets(namespace).Get(context.TODO(), daemonSetName, metav1.GetOptions{})
+	if err != nil {
+		tnf.ClaimFilePrintf("Error occurred checking for Daemonset to exist: " + err.Error())
+	}
+	// If the error is not found, that means the daemon set exists
+	return err == nil
```

Comment:
Here, you already know that err is not nil, because you tested it in line 146, so this check should be redundant. This is my suggestion for this code:
```
if err != nil {
  logrus.Errof("Error occurred checking for Daemonset to exist: %v", err)
  return false
}
return true
```